src/Support/StateIdentity.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Thunk\Verbs\Support;
+
+use InvalidArgumentException;
+use Thunk\Verbs\State;
+
+class StateIdentity
```

Comment:
There are too many different things that can point to a state. This class exists to normalize them all into one thing for ease of use around the codebase.